### ISSUE TITLE
Wrong plane count

### ISSUE THREAD

**Issue by LemonBoy** [savonet/ocaml-ffmpeg#15]
After converting a YUV420p frame into a GRAY8 one I noticed something weird, the `->linesize[1]` field is `0` but `->data[1]` is not for some reason and the resulting plane count ends up being `2` rather than `1` as expected.
The proposed fix is to change the plane counting loop into something like this
```c
for(nb_planes = 0; frame->linesize[nb_planes]; nb_planes++);
```

**Comment by gndl** [comment]
Indeed, my way of counting the planes was bad. I preferred to use the av_pix_fmt_count_planes function to get the plane count because the [AVFrame Field Documentation](http://www.ffmpeg.org/doxygen/trunk/structAVFrame.html#a1d0f65014a8d1bf78cec8cbed2304992) does not guarantee that the linesizes are properly initialized. If it does not work for you, I will use your solution.
I also deleted the frame copy functions because I realized by reading the documentation that the result can, in some cases, be wrong.
